### ISSUE TITLE
sendLocationEvent reached, but no callback

### ISSUE THREAD

**Issue by frankvollebregt** [rekab-app/background_locator#304]
Hello. First of all, sorry if this is has been asked before, and thanks for the awesome package!

I set up the package similar to the example, but my Dart callback is never called (no difference if the app is running in the foreground or background).

In the console, I do see something like the following:
```
sendLocationEvent {callback=2369190748653017841, location={...}}
```
Where all logs have the same callback identifier.

Looking through the code, this seems to be caused in the `IsolateHolderService.kt` file.
[link to the line](https://github.com/rekabhq/background_locator/blob/master/android/src/main/kotlin/rekab/app/background_locator/IsolateHolderService.kt#L272)
```kotlin
private fun sendLocationEvent(result: HashMap<Any, Any>) {
        //https://github.com/flutter/plugins/pull/1641
        //https://github.com/flutter/flutter/issues/36059
        //https://github.com/flutter/plugins/pull/1641/commits/4358fbba3327f1fa75bc40df503ca5341fdbb77d
        // new version of flutter can not invoke method from background thread

        if (backgroundEngine != null) {
            val backgroundChannel =
                    MethodChannel(backgroundEngine?.dartExecutor?.binaryMessenger, Keys.BACKGROUND_CHANNEL_ID)
            Handler(context.mainLooper)
                    .post {
                        Log.d("plugin", "sendLocationEvent $result")
                        backgroundChannel.invokeMethod(Keys.BCM_SEND_LOCATION, result)
                    }
        }
    }
```

Is there any immediate reason why this might be happening?

I tried running my code on 2 real devices:
* Google Pixel 6 (Android 12)
* Xiaomi Mi A2 (Android 10)

But both show the same behaviour.

The callback specified in `BackgroundLocator.registerLocationUpdate` is in the `LocationCallbackHandler` class as per the example:
```dart
class LocationCallbackHandler {
  ...
  
  static Future<void> callback(LocationDto locationDto) async {
    print('hello callback');
  }
}
```

Any help solving this is appreciated!

<details>
  <summary>Version information</summary>

background_locator: ^1.6.12
  
Flutter 2.8.1 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 77d935af4d (5 weeks ago) • 2021-12-16 08:37:33 -0800
Engine • revision 890a5fca2e
Tools • Dart 2.15.1
</details>

**Comment by nick1377** [comment]
1.6.12 is broken. You have to rollback to 1.6.6. The issue is actually caused by a null exception in the fromJson method in location_dto.dart due to the added provider field for location.

**Comment by RohitRB** [comment]
@nick1377  Thanks for the update. I was facing the same problem.

**Comment by RohitRB** [comment]
Does this plugin support sdk 31 (Android 12) ?

As seen in the build.gradle file it says 
compileSdkVersion 30

@mehdok 
@nick1377

**Comment by frankvollebregt** [comment]
@RohitRB I think it should, given that the most recent patch specifically mentions fixing compatibility with the Android 12 permissions.

My older test app with a 6-12 months old version of this plugin even works fine on my Android 12 Pixel 6.

**Comment by RohitRB** [comment]
@frankvollebregt 

thanks for your feedback. But i am facing issues with this new OS called FunTouch OS running android 12, 11

**Comment by RohitRB** [comment]
What can be the min interval that we can go to in seconds when keeping distance filter 1 meter so that the OS doesn't ban our app and we stop receiving location updates?

this question is a bit off track but i find no other place to put it.
@mehdok 
@atsumo 
@moovida 
@splangi 
@gomazaba

**Comment by frankvollebregt** [comment]
> 1.6.12 is broken. You have to rollback to 1.6.6. The issue is actually caused by a null exception in the fromJson method in location_dto.dart due to the added provider field for location.

Hi @nick1377, thanks for your answer. Unfortunately, it still won't work for me after rolling back the package version. Running the `example` app with 1.6.6 works, so I'll try to adapt my code to that and hope I can get it working that way.

**Comment by frankvollebregt** [comment]
I just ran the app with the `--no-sound-null-safety` flag, after which the callbacks come through as expected. This is using version 1.6.6 of the package.

**Comment by frankvollebregt** [comment]
You were completely right @nick1377, it was caused by the `provider` being `null` in the `fromJson` method. For now, I've created a fork of the repo where I set the `provider` to an empty String if it's missing. This seems to resolve the issue.

I'm not sure whether this is a solution that should be applied, or if it should instead be ensured that the provider is set: At any rate, if anyone wants to use this dependency instead for the time being, feel free:
```yaml
dependency_overrides:
  background_locator:
    git:
      url: git@github.com:frankvollebregt/background_locator.git
      ref: location-dto-null-fix
```

**Comment by frankvollebregt** [comment]
As this issue is essentially the same as #301 I'll close this one.

**Comment by daviddagan** [comment]
same issue here, please fix it as soon as you can

**Comment by RohitRB** [comment]
Please remove my email from this email

Thanks,
Rohit Rajnikant

M: 9099 20 9892 | Email: ***@***.***

**Comment by glofru** [comment]
I fix this also for Flutter 3.0: https://github.com/glofru/background_locator
